src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/FileTextExtractor.kt

```diff
@@ -0,0 +1,8 @@
+package pl.edu.uj.ii.ksi.mordor.services
+
+import java.io.File
+
+interface FileTextExtractor {
+    fun extract(file: File): String?
+    fun maxLength(maxLength: Int): FileTextExtractor
```

Comment:
Either pass it as a parameter to extract (with a default value) or create a FileTextExtractorBuilder/Factory class if you really must.